memory_reference/include/context_manager.py

```diff
@@ -0,0 +1,11 @@
+from contextlib import contextmanager
```

Comment:
Please rename the file. Context manager is the technology, which you use but not the function/task, which you implemented. I would suggest `sync_open.py`